### PR TITLE
Remove ensure activated

### PR DESCRIPTION
**Pull request by GrandSchtroumpf**
Remove the ensure activated
Transform ProxyTab into a Plugin

### CODE REVIEW

File: apps/remix-ide/src/app/panels/tab-proxy.js [hunk 1]
```diff
@@ -85,21 +92,24 @@ export class TabProxy {
...
     })
 
-    appManager.event.on('ensureActivated', (name) => {
-      if (name === 'home') {
-        // if someone force activation of home, we switch to it
-        this.event.emit('switchApp', name)
-        this._view.filetabs.activateTab(name)
-      }
-    })
+    // appManager.event.on('pluginActivated', ({name}) => {
```
Comment (LianaHus) [inline apps/remix-ide/src/app/panels/tab-proxy.js:95]: should you remove this?

File: apps/remix-ide/src/app/ui/landing-page/landing-page.js [hunk 2]
```diff
@@ -240,53 +240,43 @@ export class LandingPage extends ViewPlugin {
...
+      await this.appManager.activatePlugin(['solidity', 'source-verification'])
       this.verticalIcons.select('source-verification')
     }
-    const startPluginManager = () => {
-      this.appManager.ensureActivated('pluginManager')
+    const startPluginManager = async () => {
+      await this.appManager.activatePlugin('pluginManager')
       this.verticalIcons.select('pluginManager')
     }
+    /*
```
Comment (LianaHus) [inline apps/remix-ide/src/app/ui/landing-page/landing-page.js:267]: remove

File: apps/remix-ide/src/remixAppManager.js [hunk 3]
```diff
@@ -70,13 +70,6 @@ export class RemixAppManager extends PluginManager {
...
   }
 
-  onRegistration () {}
-
-  async ensureActivated (apiName) {
-    await this.activatePlugin(apiName)
-    this.event.emit('ensureActivated', apiName)
-  }
-
   async ensureDeactivated (apiName) {
```
Comment (LianaHus) [inline apps/remix-ide/src/remixAppManager.js:73]: and what about this?

File: apps/remix-ide/src/app/ui/landing-page/landing-page.js (hunk 2 above)
Comment (GrandSchtroumpf) [reply apps/remix-ide/src/app/ui/landing-page/landing-page.js:266]: I don't know, it was already commented. Should I ? ^^

File: apps/remix-ide/src/remixAppManager.js (hunk 3 above)
Comment (GrandSchtroumpf) [reply apps/remix-ide/src/remixAppManager.js:73]: Will work on that in another PR. I didn't wanted to be too eager ;)

File: apps/remix-ide/src/app/ui/landing-page/landing-page.js (hunk 2 above)
Comment (LianaHus) [reply apps/remix-ide/src/app/ui/landing-page/landing-page.js:266]: ah, sorry you moved it? I guess you can. it's an old workshop plugin.

**Review by LianaHus — APPROVED** [review]

**Pull request merged**